pkg/tests/end_to_end_tests/telemetry_test.go

```diff
@@ -116,12 +116,92 @@ func TestOnlyOneHousekeeper(t *testing.T) {
 
 		conn := pgxconn.NewPgxConn(db)
 
+		// Should error due to unique constraint for counter_reset_row, as there already exists a counter reset row.
 		_, err := conn.Exec(context.Background(), `INSERT INTO _ps_catalog.promscale_instance_information (uuid, last_updated, is_counter_reset_row) VALUES ('00000000-0000-0000-0000-000000000000', current_timestamp, TRUE)`)
+		require.Error(t, err)
+		require.True(t, strings.Contains(err.Error(), `violates unique constraint "promscale_instance_information_pkey"`))
+	})
+}
+
+func TestHousekeeper(t *testing.T) {
+	withDB(t, *testDatabase, func(dbOwner *pgxpool.Pool, t testing.TB) {
+		db := testhelpers.PgxPoolWithRole(t, *testDatabase, "prom_writer")
+		defer db.Close()
+
+		conn := pgxconn.NewPgxConn(db)
+
+		engine, err := telemetry.NewEngine(conn, generateUUID())
 		require.NoError(t, err)
 
-		// Should error due to unique constraint for counter_reset_row.
-		_, err = conn.Exec(context.Background(), `INSERT INTO _ps_catalog.promscale_instance_information (uuid, last_updated, is_counter_reset_row) VALUES ('00000000-0000-0000-0000-000000000000', current_timestamp, TRUE)`)
+		engine.Start()
+		defer engine.Stop()
+
+		mockMetric := prometheus.NewCounter(prometheus.CounterOpts{
+			Namespace: "test",
+			Name:      "counter",
+		})
+		require.NoError(t, engine.RegisterMetric("promscale_ingested_samples_total", mockMetric))
+
+		var val string
+		err = conn.QueryRow(context.Background(), "select value from _timescaledb_catalog.metadata where key = 'promscale_ingested_samples_total'").Scan(&val)
 		require.Error(t, err)
-		require.True(t, strings.Contains(err.Error(), `violates unique constraint "promscale_instance_information_pkey"`))
+		require.Equal(t, "no rows in result set", err.Error())
+
+		mockMetric.Add(100)
+
+		require.NoError(t, engine.Sync())
+
+		err = conn.QueryRow(context.Background(), "select value from _timescaledb_catalog.metadata where key = 'promscale_ingested_samples_total'").Scan(&val)
+		require.NoError(t, err)
+		require.Equal(t, "100", val)
+	})
+}
+
+func TestCleanStalePromscales(t *testing.T) {
```

Comment:
can we add one more new test that runs Sync for the first time when there is nothing to delete.

Reply:
Sure, but doing Sync() will add insert a row of the existing promscale, since it wasn't there earlier.